sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -24,6 +24,7 @@
     check_consistent_length,
     _check_sample_weight,
 )
+from ...utils._param_validation import Interval, Real, Integral, StrOptions
```

Comment:
`Integral` and `Real` should be imported from `numbers` where they are defined

Reply:
Thanks for the review! I will update the suggestions in new commits